pkg/core/runtime/component/resilient.go

```diff
@@ -51,7 +54,8 @@ func (r *resilientComponent) Start(stop <-chan struct{}) error {
 				r.log.WithValues("generationID", generationID).Error(err, "component terminated with an error")
 			}
 		}
-		<-time.After(backoffTime)
+		dur, _ := backoff.Next()
```

Comment:
Do we need to reset this backoff if we've worked fine for a while? I'd say if the component stays up for more than the max backoff reseting the backoff sounds good.